lib/synchronised_migration/main.rb

```diff
@@ -15,24 +15,43 @@ class << self
   end
 
   def call
-    lock_and_execute
+    done_or_execute
   end
 
   private
 
+  def done_or_execute
+    return Result.ok if previous_success?
+    result = lock_and_execute
+    mark_successful if result.success?
```

Comment:
The rest of the mark/unmark calls are in `execute`. Would this call to `mark_successful` be better placed in `execute`?